### DISCUSSION TITLE
Query Parquet file from S3

### DISCUSSION THREAD

**Discussion by ravi27may** [ClickHouse/ClickHouse#33520] [Q&A] [answered]
Is there a way to query parquet from S3 using .Net

**Comment by UnamedRus** [comment, marked as answer]
Did you look into s3(s3Cluster) table function?

https://clickhouse.com/docs/en/sql-reference/table-functions/s3/

https://altinity.com/blog/ultra-fast-data-loading-and-testing-in-altinity-cloud
https://altinity.com/blog/tips-for-high-performance-clickhouse-clusters-with-s3-object-storage

**Comment by ravi27may** [comment]
Hi @UnamedRus, thanks for the reply.

I have seen the documentation, am trying to read that from either windows cli or .net app. From the above mentioned links there is no help on how to run on those

**Reply by den-crane** [reply]
These functions are agnostic to API/OS. They are just SQL commands which you can run using any tool/API, all work will be done by a server.
